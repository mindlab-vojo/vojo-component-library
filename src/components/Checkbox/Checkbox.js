import React, { useState } from 'react'
import PropTypes from 'prop-types'
import * as Checkbox from './Checkbox.style'
import IconManager from './../IconManager'
import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
const CheckboxComponent = ({
  id,
  isChecked,
  onClickCheckbox,
  label,
  name,
  value,
  iconSize
}) => {
  const [checked, setChecked] = useState(isChecked ? true : false)
  const componentName = name ? name : id
  const handleCheckboxClick = () => {
    setChecked(!checked)
    onClickCheckbox && onClickCheckbox(checked)
  }
  const handleCheckboxChange = () => {
    // Clear warnings
  }
  const renderCheckbox = (theme) => checked ? (
    <IconManager
      icon="CheckboxChecked"
      fill={theme.colors.primaryColor}
      width={iconSize}
      height={iconSize} />
  ) : (
    <IconManager
      icon="CheckboxUnchecked"
      width={iconSize}
      height={iconSize} />
  )
  const component = (theme) => (
    <Checkbox.Wrapper>
      <Checkbox.Checkbox
        checked={checked}
        id={id}
        name={componentName}
        onChange={() => handleCheckboxChange()}
        type="checkbox"
        value={value} />
      <Checkbox.Click onClick={() => handleCheckboxClick()}>
        {renderCheckbox(theme)}
      </Checkbox.Click>
      <Checkbox.Label >
        {label}
      </Checkbox.Label>
    </Checkbox.Wrapper>
  )
  return (
    <ThemeConsumer>
      {({ theme }) => component(theme)}
    </ThemeConsumer>
  )
}
CheckboxComponent.propTypes = {
  iconSize: PropTypes.string,
  id: PropTypes.string,
  isChecked: PropTypes.bool,
  name: PropTypes.string,
  onClickCheckbox: PropTypes.func,
  label: PropTypes.node,
  value: PropTypes.node,
}
export default CheckboxComponent