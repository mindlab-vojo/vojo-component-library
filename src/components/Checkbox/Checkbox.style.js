import styled from 'styled-components';

export const Wrapper = styled.div`
  cursor: pointer;
  align-items: center;
  display: flex;
  width: 100%;
`

export const Label = styled.label`
  cursor: inherit;
  margin-left: 5px;
`

export const Checkbox = styled.input`
  display: none;
`