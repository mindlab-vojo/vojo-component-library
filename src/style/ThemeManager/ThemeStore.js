import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore