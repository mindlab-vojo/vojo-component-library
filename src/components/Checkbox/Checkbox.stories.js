import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';

import Checkbox from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const component = () => {
  const termsUrl = text('termsUrl', 'https://app.vojo.com.br/terms-of-use')
  
  return <Checkbox termsUrl={termsUrl}/>
}

export default {
  title: categories.Checkbox,
  component: component,
  decorators: [withKnobs, withA11y, (story) => <ThemeProvider>{ story() }</ThemeProvider>]
}

component.story = {
  name: 'Checkbox',
};
  