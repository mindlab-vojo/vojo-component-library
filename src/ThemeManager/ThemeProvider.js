import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer