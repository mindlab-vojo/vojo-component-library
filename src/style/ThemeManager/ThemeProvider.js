import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer