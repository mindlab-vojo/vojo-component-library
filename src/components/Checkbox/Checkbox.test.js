import React from 'react';
import { render } from '@testing-library/react';

import Checkbox from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test Checkbox component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <Checkbox
        id="checkbox"
        label="checkbox"
        name="checkbox"
        onClickCheckbox={fakeFn}
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
