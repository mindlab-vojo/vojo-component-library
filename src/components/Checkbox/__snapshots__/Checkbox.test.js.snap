// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkbox component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ fRFzZh"
      >
        <input
          class="sc-dlfnbm fKzgnC"
          id="checkbox"
          name="checkbox"
          type="checkbox"
          value=""
        />
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <label
          class="sc-gsTCUz iMnGY"
        >
          checkbox
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ fRFzZh"
    >
      <input
        class="sc-dlfnbm fKzgnC"
        id="checkbox"
        name="checkbox"
        type="checkbox"
        value=""
      />
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <label
        class="sc-gsTCUz iMnGY"
      >
        checkbox
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
